.github/workflows/chart-test.yaml

```diff
@@ -76,3 +76,66 @@ jobs:
         run: |
           echo "Some jobs failed or were cancelled."
           exit 1
+
+  deploy-to-gke:
+    runs-on: ubuntu-latest
+    env:
+      NAMESPACE: ochami-ci
+      GC_REGISTRY: ${{ secrets.GC_PROJECT_REGION }}-docker.pkg.dev/${{ secrets.GC_PROJECT_ID }}/ochami
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Configure GKE credentials
+        uses: google-github-actions/auth@v2.1.7
+        with:
+          credentials_json: ${{ secrets.GKE_SERVICE_ACCOUNT_KEY }}
+
+      - name: Set up Cloud SDK
+        uses: google-github-actions/setup-gcloud@v2
+
+      - name: Install `kubectl` plugin for gcloud CLI
+        run: gcloud --quiet components install gke-gcloud-auth-plugin kubectl
+
+      - name: Configure `kubectl` to talk to OpenCHAMI cluster in GKE
+        run: gcloud container clusters get-credentials ${{ secrets.GKE_CLUSTER_NAME }} --location ${{ secrets.GC_PROJECT_REGION }}
+
+      - name: Set up Helm
+        uses: azure/setup-helm@v4.2.0
+
+      - name: mask Google Cloud project details
+        run: echo "::add-mask::${{ secrets.GC_PROJECT_ID }}"
+
+      # I can't figure out how to embed shell commands in the `env` section of
+      # a GitHub Action, so we export it by hand here instead.
+      - name: set name of Helm installation
+        run: |
+          export HELM_INSTALL_NAME="ochami-services-$(git rev-parse --short HEAD)"
+          echo "HELM_INSTALL_NAME=${HELM_INSTALL_NAME}" >> ${GITHUB_ENV}
+
+      - name: install Helm chart
+        run: |
+          cd ./lbnl/helm;
+          helm dependency build;
+          bash ./scripts/install.sh ${HELM_INSTALL_NAME} ${NAMESPACE} ${GC_REGISTRY}
```

Comment:
How do you feel about invoking `helm install` here directly instead of using the script? That'd allow us to use `--set` parameters or a separate overlay values.yaml instead of changing the base one.

Wrapping the Helm commands in scripts prevents other tools that speak Helm (e.g. Argo) from interacting with the chart, so I'd want to avoid them if possible.

The main use case I'd expect for separate tools is for handling environment prerequisites that Helm's built-in features can't handle well. Database initialization is one such case, where it's annoying to spin up a managed cluster prior to install.

Reply:
I feel great about replacing this with `helm install` - I already don't like that load-bearing script. It exists mainly as a convenience tool to override the ~dozen container image repos without having to invoke `--set` a dozen times. Maybe there is a more Helm-ish way to do that though.

Reply:
There's not a great way to have both a default in values.yaml and an effective default from elsewhere, unfortunately. The templates like

```
{{ .Values.smd.deployment.image.repository }}/{{ .Values.smd.deployment.image.name }}:{{ .Values.smd.deployment.image.tag }}
```
can't be made to ignore the root values.yaml defaults. You can use a function that gives precedence to an optional `.Values.defaultRepository` key, but that sort of hidden behavior is worse--it won't be obvious what the effective value is unless you know the template innards.

The overlay values.yaml is equivalent to the repeated `--set`, and you can combine those if you want to keep them separate--something like `helm install /path/to/chart -f override-repositories.yaml -f override-service-type.yaml`

Reply:
> The overlay values.yaml is equivalent to the repeated --set, and you can combine those if you want to keep them separate--something like helm install /path/to/chart -f override-repositories.yaml -f override-service-type.yaml

Ah neat, I like that more. I'll break out that script and try this out instead. Thanks!